server/src/main/java/org/opensearch/cluster/routing/UnassignedInfo.java

```diff
@@ -147,7 +147,11 @@ public enum Reason {
         /**
          * Unassigned as a result of closing an index.
          */
-        INDEX_CLOSED
+        INDEX_CLOSED,
+        /**
+         * Unassigned as restored from Remote Metadata.
+         */
+        REMOTE_METADATA_RECOVERED
     }
```

Comment:
Is this shortly after quorum loss when remote cluster state restores the metadata state

Reply:
yes.

Reply:
Would the end state be the same as CLUSTER_RECOVERED from local cluster state, if not we should see what that new state would be rather than stating the mechanism of recovery

Reply:
@Bukhtawar got your point on the overhead of introducing new Reason and agree that if we can keep the same final state(`CLUSTER_RECOVERED`) for both local and remote recoveries as this makes things simpler in terms of developer and user experience.

We might be able to achieve the end result we want here without introducing this new Reason as only when state is recovered from remote we trigger RemoteStoreRecovery. When state is recovered from local disk, the recovery source is ExistingStore.

So we can add a check on Reason being `CLUSTER_RECOVERED` in StoreRecovery to skip remote recovery 

CC: @sachinpkale @gbbafna